src/subscriptions/subscriptionBackfill.ts

```diff
@@ -129,15 +135,25 @@ export function makeBackfiller(jsonRpcSenders: JsonRpcSenders) {
     if (fromBlockInclusive >= toBlockExclusive) {
       return [];
     }
-    const batchParts: BatchPart[] = [];
+    let batchParts: BatchPart[] = [];
+    const headEvents: Array<Promise<BlockHead[]>> = [];
     for (let i = fromBlockInclusive; i < toBlockExclusive; i++) {
       batchParts.push({
         method: "eth_getBlockByNumber",
         params: [toHex(i), false],
       });
+      if (batchParts.length % MAX_BATCH_SIZE === 0) {
+        headEvents.push(jsonRpcSenders.sendBatch(batchParts));
+        batchParts = [];
+      }
     }
-    const heads = await jsonRpcSenders.sendBatch(batchParts);
-    return heads.map(toNewHeadsEvent);
+    headEvents.push(jsonRpcSenders.sendBatch(batchParts));
```

Comment:
If the number of parts in the batch is a multiple of ten, then this will end up sending an empty batch.

Arguably `sendBatch` should be short-circuiting if given an empty batch, but it's not so we should either update this or `sendBatch`.

Reply:
good catch, thanks! Moved the outer `sendBatch` call into an `if` statement